### PR TITLE
reduce URL-related breadcrumbs, simplify invite unwrapping

### PR DESCRIPTION
**Pull request by trevj**
Seeing a lot of URL-related breadcrumbs in the Sentry logs. It's a bit distracting so this reduces that - mostly it was the catch block in `unwrapInvite` but I also spent a few minutes optimising it...some Friday afternoon fun. WDYT?

### CODE REVIEW

File: www/app/app.ts [hunk 1]
```diff
@@ -28,6 +28,32 @@ import {Settings, SettingsKey} from './settings';
...
+
+      // Search in the fragment for ss:// for two reasons:
+      //  - URL.hash includes the leading # (what).
+      //  - When a user opens invite.html#ENCODEDSSURL in their browser, the website (currently)
+      //    redirects to invite.html#/en/invite/ENCODEDSSURL. Since that seems like a reasonable
+      //    thing to do, let's support those URLs too.
+      const possibleShadowsocksUrl = decodedFragment.substring(decodedFragment.indexOf('ss://'));
+
+      if (new URL(possibleShadowsocksUrl).protocol === 'ss:') {
+        return possibleShadowsocksUrl;
```
Comment (bemasc) [inline www/app/app.ts]: It seems like the net effect here is to make `unwrapInvite` much less strict, and let `confirmAddServer` catch any errors.  That seems likely to work, and should achieve the goal of reducing log spam as long as `confirmAddServer` and `SHADOWSOCKS_URI.parse` don't log on failure when reading from the clipboard.

The only case I can think of where this will truly fail for us is if a server operator, using the fragment as a free text field, includes `ss://` in the server description, e.g. `ss://...#Foonet, your source for crummy ss:// links`.  If you're concerned about this case, you could add a check for `url.protocol === 'ss:'` and bail out early.

I do wonder how this change will affect our ability to evolve the invite format going forward.  For example, if a future invite is no longer compatible with old client versions (e.g. no UDP support), we could currently indicate that by using a different URL or fragment prefix.  This change would remove that capability.  Conversely, this change would allow us to create backward-compatible invites that also contain a new (e.g. non-shadowsocks) access key type, provided that the shadowsocks access key is at the end of the fragment.

File: www/app/app.ts [hunk 2]
```diff
@@ -28,6 +28,32 @@ import {Settings, SettingsKey} from './settings';
 import {Updater} from './updater';
 import {UrlInterceptor} from './url_interceptor';
 
+// If s a URL whose fragment contains a Shadowsocks URL then return that Shadowsocks URL,
```
[the remark this answers is not in the document]
Comment (trevj) [reply www/app/app.ts]: Fixed!

File: www/app/app.ts (hunk 1 above)
Comment (trevj) [reply www/app/app.ts]: Good points.

I suspect the ship has sailed on backwards compatibility, given we already have clients out there that will greedily consume any and all ss:// links. Usually there's a way to hack together some kind of backwards compatibility if you really have to (also, the app has to explicitly register the URL paths it's interested in - I see manually pasting a URL with a fragment very much as an edge case...).

**Comment by trevj** [comment]
@bemasc Thanks for the review! Your comments prodded me to add some tests for this method.
